app/src/main/java/org/traccar/client/StatusWidget.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2016 - 2022 Anton Tananaev (anton@traccar.org)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.traccar.client
+
+import android.appwidget.AppWidgetManager
+import android.appwidget.AppWidgetProvider
+import android.content.*
+import android.widget.RemoteViews
+
+import androidx.preference.PreferenceManager
+
+/**
+ * Implementation of Status Widget functionality, which can display the status of the service -
+ * whether it is enabled or not.
+ */
```

Comment:
Remove comment

Reply:
I'd keep a hint about the "status" in the comment - without examining the code, it may be not obvious which status this widget displays.
Probably a shorter comment would look better: `The widget to display the status of the service - whether it is enabled or not`?

Reply:
It's called status widget. I don't think it needs additional comment.

Reply:
The widget shows only enabled/disabled status of the service.
There are other possible statuses: running or not, online or not, geo-location obtained, etc.
The comment "limits" the responsiblity of this class.

Reply:
I've changed the way how the widget reacts on the service's status changes, so I'll remove this comment as redundant.